### DISCUSSION TITLE
home power dashboard anomoly

### DISCUSSION THREAD

**Discussion by SteveMarquess2** [David00/rpi-power-monitor#137] [Q&A]
Just noticed this recently. Running the latest (Fixes #136) power_monitor.py, the stock Home Power Dashboard is showing amps as always positive when in fact amps are going quite a bit negative during the day:

![Home_Power_Dashboard](https://github.com/user-attachments/assets/676ceaed-09f3-4fa0-bd09-e5a499b037e2)

Per earlier discussion all four CTs are "type = mains" and "reversed = false", and amps and wattage show the correct polarity in the CT Dashboards:

![Individual_CT_Data_New](https://github.com/user-attachments/assets/066103c3-0df6-4caa-bf5d-ca470575db47)
![Individual_CT_Data_New-amps](https://github.com/user-attachments/assets/2dfd18d0-040e-49ec-b681-409afd66bd8a)

I've not tried to dig into this yet, just throwing out this query in the hope it's something simple.

FWiW the Daily Overview in the Home Power Dashboard (not shown here) doesn't look right; on all but the rainiest of days (I wish, we need rain now) this premises is considerably overproducing at this time of year. Note I have not calibrated amperage for the CTs and so haven't worried about quantitative accuracy yet. Once I've calibrated and figured out the Home Power amperage issue I'll compare the Daily Overview to actual daily net meter readings.

**Comment by David00** [comment]
Hi Steve, the amperage graph in the Home Power Dashboard is always going to be positive - this is intended to be the measure of amps that the home is using. 

The net utilization - while it doesn't show amperage (by default) - will go positive/negative.  The net utilization graph will show the total that is being sent to/from the grid.

While the net figures are pretty straight forward, the "Home Power Usage" and "Amperage" charts will not make sense without measuring the actual outputs of the production sources with a dedicated CT.  The logic is essentially:

If you're seeing, say, 40A go out your mains (a negative reading), then the house/load is _at least_ 40A.  The home power watts graph technically shouldn't be negative as this is a measure of what the house is using, but without tracking the individual sources of power generation (or every single breaker in the panel) it's not possible to calculate.

Hopefully this helps!

**Reply by SteveMarquess2** [reply]
Ummm, that doesn't make sense. If I'm exporting 9600W (40A) at my service entrances (which can and does happen given I currently have 14KW of load side backfeeds), then:

1) The premises (load side of service disconnects) is NOT consuming 9.6KW!! Actual loads are unknowable from those four CTs. All we know is that the excess of PV production over loads is 9.6KW. In practice it will be something like 2-4KW loads and 12-14KW production, for a net ~10KW export to the grid. The POCO does not see those internal loads and neither do your four CTs; those loads are unknowable at the service entrances (SEs).

2) Amps should be negative as power is flowing FROM the"load" side TO the grid. There is a big difference between importing/consuming 9.6KW (positive amps) and exporting 9.6KW (negative amps). IMHO it is correct and proper to say that the premises is consuming negative watts (and amps) when exporting. That makes perfect sense to me even (especially) absent any RPi Power Monitor knowledge of the PV production. By design my system spends a lot of time fluctuating around net zero. Negative versus positive matters. Negative fluctuations mean I'm golden during (all too frequent) grid outages. Positive fluctuations (amps specifically) mean I must draw down the battery bank and use the autostart diesel generator. I'll be replacing the battery bank and so am very interested in that positive versus negative amp flow over time at the SEs.

3) To complicate the situation even more, note that in addition to the 14KW of load side backfeeds I also have 3KW of supply side backfeed (supply side meaning upstream of the service disconnects and thus invisible to the four "mains" CTs). Eventually I plan to track all three sets of PV interconnection, but independently of the true power flow across the SEs.

**Reply by David00** [reply]
Yes, you're absolutely right! I was multitasking when responding and that led to some faulty logic being included in my response.

The "home power" calculation is actually quite simple, but again, it relies on having a measure of circuits where you have energy production. Without any `type = production` CTs, it's impossible to know how much of that power you're actually consuming at any given time.

Assuming the signage is correct (production channels positive when producing, and mains channels positive when consuming from the grid), it will all work out. 

> Amps should be negative as power is flowing FROM the"load" side TO the grid. There is a big difference between importing/consuming 9.6KW (positive amps) and exporting 9.6KW (negative amps). IMHO it is correct and proper to say that the premises is consuming negative watts (and amps) when exporting. 

The Net Utilization graph is the one tracking the import/export at the service entrance. It will show positive or negative current when importing from the grid or exporting back to it, respectively. If you want to add the amperage to the Net Utilization panel in Grafana, I've built it for you here - just pull up the dashboard (as admin), select the 3-dot menu for the Net Utilization panel, select Inspect -> Panel JSON, and then delete the entire panel JSON text and replace it with the contents of this file:
[Net Utilization Panel (with Amperage).txt](https://github.com/user-attachments/files/16463271/Net.Utilization.Panel.with.Amperage.txt)

Your point 3 does sound complicated!

**Reply by SteveMarquess2** [reply]
> The "home power" calculation is actually quite simple, but again, it relies on having a measure of circuits where you have energy production. Without any `type = production` CTs, it's impossible to know how much of that power you're actually consuming at any given time.

In time I will add CTs for the three separate points of PV backfeed. In the interim I have (for two such inputs) the realtime proprietary vendor data (which I can also extract via Home Assistant). I'm a lot less interested in a "home power" calculation than I am in tracking  actual net utilization at the service point (which note is _not_ the same thing as your home power calculation) and PV production.

> If you want to add the amperage to the Net Utilization panel in Grafana, I've built it for you here - just pull up the dashboard (as admin), select the 3-dot menu for the Net Utilization panel, select Inspect -> Panel JSON, and then delete the entire panel JSON text and replace it ...

Much appreciated! I know I need to dig into Grafana enough to get my bearings but this will be a great study aid. I gave that a try (in a duplicate of the Home Power Dashboard) and the JSON code quickly reverts back to the original. Probably basic Grafana ignorance on my part; the "Apply" instead of "Save" is probably a clue. 

> Your point 3 does sound complicated!

Yup. My first PV system is thirteen years old, from before the time you could get integrated systems from vendors. IIRC I had to integrate products from eight different vendors to get it all working. I have to use both Ethernet and RS485 to talk to two proprietary (and buggy) vendor interfaces to get performance data. Recently I added a set of Enphase microinverters (which theoretically has an Ethernet accessible UI) and discovered that their UI is essentially worthless without real time direct Internet access which isn't going to happen. So I'm going to ignore their worthless UI entirely and just use your power monitor instead. I've been tracking everything on a monthly basis on multiple conventional KWh meters, but it's going to be nice to be able to see all the production and net meter data in real time in one place.

I have a couple hundred feet of conduit to trench before I can get comm lines out to the ground mount structures where the CTs need to go. I'll get that done before winter, hopefully.
